generator/views/ping_view.py

```diff
@@ -42,7 +42,10 @@ def from_db_views(
 
                 if channel.get("channel") is not None:
                     table["channel"] = channel["channel"]
-                if len(references) != 1 or references[0][-2] != f"{dataset}_stable":
+                if len(references) != 1 or references[0][-2] not in [
+                    f"{dataset}_stable",
+                    dataset,
```

Comment:
Wouldn't the change in `namespaces.py` set result in this being the app dataset, e.g. `fenix` (which won't be referenced by the view)?

Thinking ahead a little more, at some point when we'll have the `UNION` app dataset views, this condition won't hold anymore since we'll be referencing multiple tables. So adding something like `ping_table: true` to the metadata is probably the most convenient path forward

Reply:
> Wouldn't the change in `namespaces.py` set result in this being the app dataset, e.g. `fenix` (which won't be referenced by the view)?

No, because the `dataset` variable here is referring to the dataset we extracted out of the references, which *is* the app-id-specific dataset (e.g. `org_mozilla_firefox`).

Reply:
I'm not sure what I was thinking when I wrote the above, but I was wrong and the above code doesn't work. I added a new concept called a `source_dataset` (the app-id-level view for the case of the release channel, the stable table in the case of non-release) which we can use here to work around this for now.